### ISSUE TITLE
Bug in TTestPower()

### ISSUE THREAD

**Issue by JingyiShi** [statsmodels/statsmodels#9378]
The setting "one-sided" suggested in the embed document does not work. I've tried "larger" and "smaller" which were suggested in the webpage document; only "larger" worked, but "smaller" does not work. I am using Statsmodels 0.14.0 and Jupyter Lab 4.0.11 with Python 3.11.7.

```python
from statsmodels.stats.power import TTestPower
TTestPower().solve_power(effect_size=0.5, alpha=0.05, power=0.8, alternative='one-sided')
```

Error message is long so I am not copy/pasting here.

Thanks!

**Comment by josef-pkt** [comment]
I don't think we have any regular power function that have alternative with "one_sided".
There is "one-sided" alternative for the shortcut power functions which ignore the opposite tail. (AFAIR)

result is kind of what I expect.

except that the return should be nan or raise an exception for the **impossible** case, as in this case when the alternative goes in the wrong direction. (There is no number of observations that can have the required power and significance alpha, power is always smaller than alpha)
I need to check the details for this case again.

```
TTestPower().solve_power(effect_size=0.5, alpha=0.05, power=0.8, alternative='larger')
                     
26.137504465263437

TTestPower().solve_power(effect_size=0.5, alpha=0.05, power=0.8, alternative='smaller')

Warning (from warnings module):
  File "...\statsmodels\statsmodels\stats\power.py", line 524
    warnings.warn(convergence_doc, ConvergenceWarning)
ConvergenceWarning: 
Failed to converge on a solution.

array([10.])
TTestPower().solve_power(effect_size=-0.5, alpha=0.05, power=0.8, alternative='smaller')
                     
26.137504465263437
```

**Comment by josef-pkt** [comment]
We should find explicit solution to intercept impossible cases at least for computing sample size. I'm not sure about cases when we solve for other variables.

For one-sided alternatives we could just do a sign check on effect size.
requirement for t and z-tests
for alternative is larger: effect > 0
for alternative is smaller: effect < 0
(AFAICS, current function in `power.py` do not have a `value` option, i.e. nonzero null hypothesis have to be incorporated in computation of effect_size

same point in #9108

F-test and chisquare test currently have no `alternative` option.
e.g. we cannot use it for ratio of 2-sample variance hypothesis test which would need ratio < 1, or ratio > 0

**Comment by josef-pkt** [comment]
doc bug:
https://www.statsmodels.org/dev/generated/statsmodels.stats.power.TTestPower.solve_power.html#statsmodels.stats.power.TTestPower.solve_power
 `alternative` doc says "one-sided"
`power` method has correct larger/smaller alternative

power docstrings need review
also style: capitalization, punctuation

FTestPower.solve_power.html ncc docstring mentions see Notes, but it does not have the corresponding explanation in notes section, `power` method does have it.

Note: solve_power method is only added to subclasses to have explicit keywords and docstring
